pkg/provider/hpa.go

```diff
@@ -197,20 +201,51 @@ func (p *HPAProvider) updateHPAs() error {
 		p.collectorScheduler.Remove(ref)
 	}
 
-	p.logger.Infof("Found %d new/updated HPA(s)", newHPAs)
+	if newHPAs > 0 {
+		p.logger.Infof("Found %d new/updated HPA(s)", newHPAs)
+	} else {
+		p.logger.Debug("No new/updated HPAs found")
+	}
+
 	p.hpaCache = newHPACache
 
 	return nil
 }
 
 // equalHPA returns true if two HPAs are identical (apart from their status).
 func equalHPA(a, b autoscalingv2.HorizontalPodAutoscaler) bool {
-	// reset resource version to not compare it since this will change
-	// whenever the status of the object is updated. We only want to
-	// compare the metadata and the spec.
-	a.ObjectMeta.ResourceVersion = ""
-	b.ObjectMeta.ResourceVersion = ""
-	return reflect.DeepEqual(a.ObjectMeta, b.ObjectMeta) && reflect.DeepEqual(a.Spec, b.Spec)
+	return annotationsUpToDate(a.ObjectMeta, b.ObjectMeta, kubectlLastAppliedAnnotation) && reflect.DeepEqual(a.Spec, b.Spec)
+}
```

Comment:
Do we want a test-case for this? Or are the current test-cases enough?

Reply:
Added test case!